src/core/api/link_raw_api.cpp

```diff
@@ -444,6 +444,15 @@ void LinkRaw::InvokeTransmitDone(otRadioFrame *aFrame, bool aFramePending, otErr
 #endif
 }
 
+#if OPENTHREAD_CONFIG_LEGACY_TRANSMIT_DONE
+void LinkRaw::InvokeTransmitDone(const otRadioFrame *aFrame, bool aFramePending, otError aError)
+{
+    // A trick to indicate whether FramePending is set.
+    const otRadioFrame *ackFrame = aFramePending ? aFrame : NULL;
```

Comment:
Good idea, I will generate this ACK frame in those platforms not supporting passing ACK frame yet.